src/request.ts

```diff
@@ -77,27 +77,59 @@ export async function buildRequestUrl(
 }
 
 /**
- * The error type returned by `executeRequestError`.
+ * The error type returned by the SDK whenever it makes a network request
+ * (internally, this happens in `executeRequest`). It implements, so is
+ * compatible with, `AxiosError`.
  */
-export class ExecuteRequestError extends Error {
-  originalError: AxiosError;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class ExecuteRequestError<T = any, D = any> extends Error implements AxiosError {
   responseStatus: number | null;
   responseMessage: string | null;
 
+  // Properties required by `AxiosError`.
+
+  config: AxiosRequestConfig<D>;
+  code?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  request?: any;
+  response?: AxiosResponse<T, D>;
+  isAxiosError: boolean;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  toJSON: () => object;
+
   /**
    * Creates an `ExecuteRequestError`.
    *
    * @param message - The error message.
-   * @param axiosError - The original axios error.
+   * @param axiosError - The original Axios error.
    * @param responseStatus - The response status, if found in the original error.
    * @param responseMessage - The response message, if found in the original error.
    */
   constructor(message: string, axiosError: AxiosError, responseStatus: number | null, responseMessage: string | null) {
+    // Include this check since ExecuteRequestError implements AxiosError, but
+    // we only expect original errors from Axios here.
+    if (axiosError instanceof ExecuteRequestError) {
```

Comment:
Since we expect it to always be an `AxiosError`, wouldn't it be better to have a check like this here?
```js
if (!(axiosError instanceof AxiosError)) {
    ...
}
```

 Right now we could pass any non-`ExecuteRequestError` here and there'd be no complaint.

Reply:
Unless I'm misunderstanding you, that case would be caught by the typing given above for `axiosError`. However, that typing doesn't exclude an `ExecuteRequestError` being passed in to the constructor. I made `ExecuteRequestError` a subtype of `AxiosError` for API compatibility (as consumers might be relying on the SDK to throw `AxiosError`). But, `ExecuteRequestError` being passed here is almost certainly a developer/logic bug. 

Hope that makes sense! I updated the comment here, but if it's still a source of confusion maybe I can just remove this check.